### PR TITLE
Use "treeList" finder if model has TreeBehavior loaded.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by jippi** [comment]
icanhaztest?

**Comment by ADmad** [comment]
@jippi Hows that look?

File: src/Listener/RelatedModelsListener.php [hunk 1]
```diff
@@ -51,7 +52,8 @@ public function publishRelatedModels($action = null)
                 continue;
             }
 
-            $query = $association->find()->find('list');
+            $finder = $this->_finder($association);
```
Comment (deizel) [inline src/Listener/RelatedModelsListener.php]: Undefined method - was this intended to be protected, or public like in the tests?

Comment (ADmad) [inline src/Listener/RelatedModelsListener.php]: @deizel It should be public. I have pushed a fix.

Comment (deizel) [inline src/Listener/RelatedModelsListener.php]: :+1:

**Pull request merged**